ui/widgets/dialog.js

```diff
@@ -347,7 +347,8 @@ $.widget( "ui.dialog", {
 
 				// Setting tabIndex makes the div focusable
 				tabIndex: -1,
-				role: "dialog"
+				role: "dialog",
+				"aria-modal": "true"
```

Comment:
If @fnagel agrees with my suggestion then this would need to be changed to only be set when `this.options.modal` is truthy. Since setting `"false"` is apparently equivalent to omitting the attibute, we should be able to just check this condition here:
```suggestion
				"aria-modal": this.options.modal ? "true" : "false"
```
We'd also need a test for both cases.

Reply:
does that mean using the dialog with the modal option set to false is for cases integrating the div within a page while with the modal option set to true the dialog gets the modal behavior as an overlay? i've just tested your suggested change on an instance of drupal 11 and it is still a drop in change with no further changes necessary. so i would agree. i was just simply unaware of that modal option. but let's wait what @fnagel says. if he agrees i'll commit your suggestion.

Reply:
Dialog traps focus - if you press TAB while the last tabbable element has focus, it focuses the first element in the dialog. If you have SHIFT pressed (so tabbing is done in reverse) and the first element is focused, it focuses the last focusable one:
https://github.com/jquery/jquery-ui/blob/9180a8180b17c38f6c3f27ba46d4546d800d3508/ui/widgets/dialog.js#L364-L384

This happens regardless of the `modal` option, perhaps a bit confusingly - you can test on https://jqueryui.com/resources/demos/dialog/animated.html. But you can still click on elements outside, it's just the TAB key that's blocked. I'm not sure that's intended but it does seem to suggest we shouldn't add `aria-modal` to such dialogs.

The overlay only appears if the `modal` flag is truthy and it's done in the `_createOverlay` method here:
https://github.com/jquery/jquery-ui/blob/9180a8180b17c38f6c3f27ba46d4546d800d3508/ui/widgets/dialog.js#L875-L877

That method checks the `modal` flag at the beginning:
https://github.com/jquery/jquery-ui/blob/9180a8180b17c38f6c3f27ba46d4546d800d3508/ui/widgets/dialog.js#L845-L848

This leads me to thinking the `aria-modal` attribute should reflect the `modal` option.

Reply:
BTW, I'd tweak my suggestion - no need to set `"false"` when we can pass `null`; that's equivalent to removing the attribute:
```suggestion
				"aria-modal": this.options.modal ? "true" : null
```

Reply:
Interesting question that Michał opened here. I'm not quite sure on how to use it as I'm not really familiar with the `aria-modal` attribute but when reading some resources, there are some hints that the attribute should be used when its not possible to interact with the content outside the dialog (sometimes even mentioning an overlay / dimmer in the background). In some it is formulated quite clearly: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/ (see the "note" box at the bottom.

In my experience, all ARIA attributes can behave different for specific browser / AT setups but to me it looks like we should use the the approach @mgol suggested. Just to be sure we do not break stuff for some users.